### PR TITLE
Modified code in client.go to avoid writing to an instance of skipLoggingPayload to avoid potential data race condition

### PR DESCRIPTION
**Pull request by shrsr**

### CODE REVIEW

File: client/client.go [hunk 1]
```diff
@@ -428,10 +428,54 @@ func (c *Client) MakeRestRequest(method string, rpath string, body *container.Co
 	return req, nil
 }
 
+func (c *Client) makeRestRequestRaw(method string, rpath string, payload []byte, authenticated bool) (*http.Request, error) {
```
Comment (samiib) [inline client/client.go:431]: Is there a way to fix the race condition without duplicating the function?
eg. Change `MakeRestRequestRaw` so it calls `makeRestRequestRaw` and `skipLoggingPayload` can be a parameter of `makeRestRequestRaw`. Then `MakeRestRequestRaw` can just pass in `c.skipLoggingPayload` and the Authenticate function can pass in True for skip logging. Would that work?

**Review by samiib — CHANGES_REQUESTED** [review]

File: client/client.go (hunk 1 above)
Comment (shrsr) [reply client/client.go:431]: Yes, the next comes removing the repetitive code. I quickly made this PR so that I had the changes saved somewhere.

Comment (samiib) [reply client/client.go:431]: Aha cool, thanks! I'll test it shortly and let you how it goes.

**Comment by lhercot** [comment]
Closing this PR as PR #294 has been merged to fix this issue.

**Pull request closed without merging**